doc/source/reporting.rst

```diff
@@ -82,17 +82,19 @@ Others:
 
       - Provide an alias from one *key* to another:
 
-        >>> r.add('aliased name', 'original name')
+        >>> from message_ix.reporting import Reporter
+        >>> rep = Reporter()  # Create a new Reporter object
+        >>> rep.add('aliased name', 'original name')
 
       - Define an arbitrarily complex computation in a Python function that
         operates directly on the :class:`ixmp.Scenario`:
 
         >>> def my_report(scenario):
         >>>     # many lines of code
         >>>     return 'foo'
-        >>> r.add('my report', (my_report, 'scenario'))
-        >>> r.finalize(scenario)
-        >>> r.get('my report')
+        >>> rep.add('my report', (my_report, 'scenario'))
+        >>> rep.finalize(scenario)
+        >>> rep.get('my report')
         foo
 
       .. note::
```

Comment:
please @khaeru confirm what you meant here with values (2). Was (2) a reference? 

Thanks.

Reply:
@francescolovat good question.

Basically, the `.. automethod: add` prints the docstring in the Python code first: https://github.com/iiasa/ixmp/blob/480d0d81e74e303dd2e3ef1aed3007abbfc3618a/ixmp/reporting/__init__.py#L208-L227 …followed by the additional documentation in this .rst file. So if you look at this rendered HTML, line 101 appears after the list where it says “2. any other literal value or constant.”

To explain further:
```
>>> rep.add('y', 'x')
>>> rep.get('y')  # returns the string 'x'
'x'
>>> rep.add('x', 42)  # now 'x' above will be interpreted as a reference to this new node 'x'
>>> rep.get('y')
42
```

This is why [`partial()`](https://docs.python.org/3/library/functools.html#functools.partial) is used at many places in the code. Also, I am summarizing [this part of the dask documentation](https://docs.dask.org/en/latest/spec.html).

Again, your suggestion on how to make this more clear very much appreciated.

Reply:
![image](https://user-images.githubusercontent.com/56022262/70507296-595d0200-1b2c-11ea-8902-3bc541a9c4dc.png)

I was asking because unfortunately line 101 does not appear right after “_2. any other literal value or constant._”, it is indeed located far below where the reader loses the perception that that `(2)` actually refers to the second point of the description of the _computation_ argument. I've committed a suggestion to make the note more clear. I hope it will be OK, @khaeru, after that, I think this PR is ready to merge 😃 .

Reply:
I like the new version! Thanks.